src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -66,6 +66,9 @@ public JsonTypeInfo BaseJsonTypeInfo
         // Whether to use custom number handling.
         public JsonNumberHandling? NumberHandling;
 
+        // Required properties left
+        public HashSet<JsonPropertyInfo>? RequiredPropertiesLeft;
```

Comment:
Related to https://github.com/dotnet/runtime/pull/72937#discussion_r931487606, using `HashSet` to track one or two required properties might be overkill. It might make sense to use a dedicated data structure that falls back to `HashSet` if the numbers of properties becomes substantial (in the style of `JsonPropertyInfoValues`).

It might start as a single `byte` that holds state for up to 8 required properties, which then falls back to a `HashSet` or `BitArray`.

Reply:
I'll file an issue to optimize this but I prefer to avoid any custom data structures for RC since it's much more error prone. Let's keep it simple for now

Reply:
Calling it a custom data structure is probably a misnomer, I meant using a wrapper type (which is what JsonPropertyInfoValues is doing). Given the discussion, I don't believe HashSet is the appropriate data structure for this work.

Reply:
I ended up using BitArray